### ISSUE TITLE
Web - iOU - Cancel request message wrongly stays in chat and iOU preview until signing/signout

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#17273]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Navigate to any conversation with an user you have access to and Request $10
2. Disable the internet connection on the main testing device.
3. As the secondary user/device navigate to the conversation with the main user where the 10$ were just requested
4. As the Secondary user/device Pay the IOU via "Settled up elsewhere"
5. On the main testing device while offline - cancel the 10$ IOU
6. Enable the internet connection on the main testing device
7. Click/tap the X to dismiss the error message

## Expected Result:
Cancel request message should disappear from chat and iOU preview

## Actual Result:
Cancel request message wrongly stays in chat and iOU preview until signing/signout

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.2.98.2

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**
https://expensify.testrail.io/index.php?/tests/view/3347999

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/231218336-75aea1fd-c056-4a50-96ef-afda4757755f.mp4

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @zanyrenney (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01b092c29049b9b824

**Comment by MelvinBot** [comment]
Current assignee @zanyrenney is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @aimane-chnaif (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @iwiznia (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by zanyrenney** [comment]
Oops, did not mean to apply this label.

**Comment by zanyrenney** [comment]
I cannot reproduce this bug @kbecciv - can you confirm this is still happening please?

**Comment by zanyrenney** [comment]
#15297 should fix this one so I think we should hold off until we have the solution implemented there to see if this is fixed.

**Comment by zanyrenney** [comment]
Still waiting on progress in 15297.

**Comment by zanyrenney** [comment]
Latest update on other issue: Daily update: my proposal is under our internal review in the Callstack team. Should be posted here soon.

**Comment by zanyrenney** [comment]
Still on HOLD https://github.com/Expensify/App/issues/15297#issuecomment-1560270064

**Comment by zanyrenney** [comment]
> Still held on https://github.com/Expensify/App/issues/19066, but nice to see we've identified a solution there so things are moving

from the main issue!

**Comment by zanyrenney** [comment]
as mentioned by Carlos here - https://github.com/Expensify/App/issues/19066#issuecomment-1593735387
This is still being worked on there so we can't move forward here yet.

**Comment by zanyrenney** [comment]
Latest is that Carlos was OOO but is actively working on this. Bug0 assignee gave and update here about this https://github.com/Expensify/App/issues/19066#issuecomment-1617914847

**Comment by zanyrenney** [comment]
All PRs are in review. This has been a tricky one, lots of moving parts - https://github.com/Expensify/App/issues/19066#issuecomment-1624024031

**Comment by zanyrenney** [comment]
Still on HOLD>

**Comment by zanyrenney** [comment]
https://github.com/Expensify/App/issues/19066 this is now live.

**Comment by zanyrenney** [comment]
I now cannot reproduce this. Closing!
